src/util/tpool/fd_tpool.cxx

```diff
@@ -29,12 +30,21 @@ fd_tpool_private_worker( int     argc,
   FD_VOLATILE( worker->state ) = FD_TPOOL_WORKER_STATE_BOOT;
   FD_COMPILER_MFENCE();
 
+  if( scratch_sz ) {
+    if( !scratch ) {
+      scratch = aligned_alloc( FD_SCRATCH_SMEM_ALIGN, fd_ulong_align_up( scratch_sz, FD_SCRATCH_SMEM_ALIGN ) );
```

Comment:
Is this change necessary? Can't the caller do the aligned alloc? Having tpool take an implicit / hidden dependency on malloc is the kind of stuff we have hated about Rust.

Reply:
Worker threads need to allocate scratch space and this is the only way to do it.